### PR TITLE
Update to 1.20.6 and java 21, refactor small methods

### PR DESCRIPTION
**Pull request by MajdT51**
Tested on local server with paper 1.20.6

If you agree with the changes, then tell me please and I can also test it again on my main server before the merge.

continuous-integration/appveyor/pr  checks probably failed due to java 17, we probably need 21.

Changes:
- Update to java 21
- Update to 1.20.6
- Add new blocks and materials
- replace GRASS with SHORT_GRASS
- A Sign has now 2 sides, select front side to write
- Replace deprecated setEditable() method with setWaxed()
- Refactor teleport player task
- Replace deprecated creature.getPassenger() with creature.getPassengers()
- Check if a parrot is sitting before teleporting
- Add information about the location of regionpost for command /whichregion - thanks to @gnubufferoverflows

### CODE REVIEW

**Comment by RoboMWM** [comment]
seems the appveyor.yml has to be updated. Maybe make that a separate PR? I think a few of my projects have been updated for java 21 (I think at least GriefPrevention? though I'd have to doublecheck.)

**Comment by MajdT51** [comment]
> seems the appveyor.yml has to be updated. Maybe make that a separate PR? I think a few of my projects have been updated for java 21 (I think at least GriefPrevention? though I'd have to doublecheck.)

the image supports java21, I had only to change the jdk version. I marked the PR as draft because I didn't test it on my public server yet. When you give me the OK then I would do that :)

**Comment by RoboMWM** [comment]
The standard is you test your own changes lol. I guess I see what you mean... draft PR usually means you have some work to do on it before it's ready to be reviewed tho lol. But it's fine yes.

**Comment by RoboMWM** [comment]
The only thing I do recommend though is splitting up your changes into multiple PRs. Don't have to do it now, but something to consider if you don't intend on maintaining this plugin.

**Comment by MajdT51** [comment]
> The standard is you test your own changes lol. I guess I see what you mean... draft PR usually means you have some work to do on it before it's ready to be reviewed tho lol. But it's fine yes.

@RoboMWM thank you, I tested it ofc on a local server while updating it, and **today** on my public server (paper 1.20.6) with other players. Everything went well, We didn't notice anything wrong.

It could be now reviewed, merged and released. About maintaining it by myself, I'm interested but I still have some simple questions to clarify. I will chat with you on discord :) 

Thanks

File: pom.xml [hunk 1]
```diff
@@ -112,12 +118,6 @@
             <artifactId>guava</artifactId>
             <version>19.0</version>
         </dependency>
-        <!--json stuff-->
-        <dependency>
-            <groupId>com.googlecode.json-simple</groupId>
-            <artifactId>json-simple</artifactId>
-            <version>1.1.1</version>
-        </dependency>
```
Comment (RoboMWM) [inline pom.xml:120]: Hmm, are you sure this isn't needed? Not sure if this is leftover from me copying the pom for GriefPrevention long ago, or if it's still needed. From what I recall, I observe projects building if these exist in either the target directory or maven cache.

File: src/me/ryanhamshire/PopulationDensity/RegionCoordinates.java [hunk 2]
```diff
@@ -53,42 +53,39 @@ public static RegionCoordinates fromLocation(Location location)
...
     @Override
-    public boolean equals(Object coordinatesToCompare)
-    {
+    public boolean equals(Object coordinatesToCompare) {
         if (coordinatesToCompare == null) return false;
 
         if (!(coordinatesToCompare instanceof RegionCoordinates)) return false;
 
-        RegionCoordinates coords = (RegionCoordinates)coordinatesToCompare;
+        RegionCoordinates coords = (RegionCoordinates) coordinatesToCompare;
```
Comment (RoboMWM) [inline src/me/ryanhamshire/PopulationDensity/RegionCoordinates.java]: minor style thing, but seems inconsistent, as far as I know, for explicit casts there is no space.

File: src/me/ryanhamshire/PopulationDensity/ScanRegionTask.java [hunk 3]
```diff
@@ -22,44 +22,158 @@
...
+    ));  
+
+    public ScanRegionTask(ChunkSnapshot[][] chunks, boolean openNewRegions) {
         this.chunks = chunks;
         this.openNewRegions = openNewRegions;
     }
 
-    private class Position
-    {
+    private class Position {
```
Comment (RoboMWM) [inline src/me/ryanhamshire/PopulationDensity/ScanRegionTask.java]: I prefer newline braces, but, is easier if style changes are left for a separate PR.

**Review by RoboMWM — COMMENTED** [review]
I see, a lot of this PR is mostly formatting changes. Unless you're willing to maintain, I'd prefer these to at least be separated so it's easier to differentiate between actual code changes vs. running an automated style formatter.

File: src/me/ryanhamshire/PopulationDensity/ScanRegionTask.java (hunk 3 above)
Comment (MajdT51) [reply src/me/ryanhamshire/PopulationDensity/ScanRegionTask.java]: I will try to remove back all styling changes in another commit.

File: src/me/ryanhamshire/PopulationDensity/RegionCoordinates.java (hunk 2 above)
Comment (MajdT51) [reply src/me/ryanhamshire/PopulationDensity/RegionCoordinates.java]: okay, I'm using the default styler from IntelliJ and used to whitespaces after ")" and add it unintentionally when i see it. But I will reset the styling to keep it consistent.

File: pom.xml (hunk 1 above)
Comment (MajdT51) [reply pom.xml:120]: I don't see that we are using this json library anywhere. It was also removed from GriefPrevention in later versions.

**Comment by MajdT51** [comment]
thank you for the review. I undone my own styling to keep it consistent. I agree it was unfortunately a big not good planed PR which is not easy to review and test. I will make sure to keep PRs small in the future even if I'm the one who maintaining it.

**Review by RoboMWM — APPROVED** [review]

**Pull request merged**